File: Java/src/main/java/org/codecop/dependencies/extract_and_override_call/Checkout.java
package org.codecop.dependencies.extract_and_override_call;

public class Checkout {

    public Receipt createReceipt(Money amount) {
        Receipt receipt = new Receipt();
        Money vat = amount.percentage(20);

        receipt.setAmount(amount);
        receipt.setTax(vat);
        receipt.setTotal(amount.add(vat));

        ReceiptRepository.store(receipt);

        return receipt;
    }
}

File: CSharp/DependencyBreakingKatas/ExtractAndOverrideCall/Checkout.cs
namespace Org.Codecop.Dependencies.ExtractAndOverrideCall
{
    public class Checkout
    {
        public Receipt CreateReceipt(Money amount)
        {
            var receipt = new Receipt();
            var vat = amount.Percentage(20);

            receipt.Amount = amount;
            receipt.Tax = vat;
            receipt.Total = amount.Add(vat);

            ReceiptRepository.Store(receipt);

            return receipt;
        }
    }
}
